packages/mgt-chat/src/components/ChatList/ChatList.tsx

```diff
@@ -121,6 +114,47 @@ export const ChatList = ({
     }
   }, [chatListClient]);
 
+  useEffect(() => {
+    // when chat threads are updated, add markAllAsRead callback to menuItems
+    const chatThreads = chatListState?.chatThreads;
+    if (chatThreads && chatThreads.length > 0) {
+      log("chatThreads updated, adding 'mark all as read' to menu...");
+      const markAllAsRead = {
```

Comment:
can you explain why we have to reset the markAllAsRead menu item each time VS it is declared once when the component is loaded?

Reply:
Good question. onloaded wasn't present until I merged latest PR into this branch. I'll update.